--- primitives.lisp	2010-09-16 17:26:08.123999999 +0200
+++ primitives.lisp.new	2010-09-16 17:58:33.089999985 +0200
@@ -89,6 +89,7 @@
           *run-or-raise-all-screens*
           *deny-map-request*
           *deny-raise-request*
+          *deny-quit-request*
           *suppress-deny-messages*
           *honor-window-moves*
           *resize-hides-windows*
@@ -843,6 +844,9 @@
 
 Note that no denial message is displayed if the window is already visible.")
 
+(defvar *deny-quit-request* nil
+ "Deny quitting a window")
+
 (defvar *suppress-deny-messages* nil
   "For complete focus on the task at hand, set this to @code{T} and no
 raise/map denial messages will be seen.")
